tidy3d/components/medium.py

```diff
@@ -93,8 +140,23 @@ class AbstractMedium(ABC, Tidy3dBaseModel):
         "useful in some cases.",
     )
 
+    nonlinear_spec: NonlinearSpecType = pd.Field(
+        None,
+        title="Nonlinear Spec",
+        description="Nonlinear spec applied on top of the base medium properties.",
+    )
+
     _name_validator = validate_name_str()
 
+    @pd.validator("nonlinear_spec", always=True)
+    def _validate_nonlinear_spec(cls, val):
+        """Validate compatibility with the nonlinear spec."""
+        if val is None:
+            return None
+        if issubclass(cls, Medium):
```

Comment:
I think this is a problematic way to do this. It's a bit circular? if we moved `Medium` and `AbstractMedium` to their own files, it would break I think. We're dealing with some issues like this in `geometry.py` right now and it's a headache so I'd say let's keep AbstratMedium agnostic to Medium.

A few possible solutions:

1. only move `nonlinear_spec` to the `Medium` for now until we find a better way to do this.

2.  add `nonlinear_spec` to `AbstractMedium` like this, but only allow `None` values.  if it's not None, give the warning that you have here (optional).  Add `nonlinear_spec`  to `Medium` with a allowed type of `Union[Chi3Spec]` and overwrite validator if we had one.

In general, we need a simple way to validate whether we support different possible combinations of specs.  Haven't figured out a good solution yet.